### ISSUE TITLE
Code in datasets.py

### ISSUE THREAD

**Issue by ling-cai** [facebookresearch/tkbc#2]
in datasets.py  'while id_timeline < len(self.events) and self.events[id_timeline][0] <= eval_events[id_event][3]:' should it be 'eval_events[id_event][0]' instead of 'eval_events[id_event][3]'?

**Comment by timlacroix** [comment]
hmm, I may be wrong because this code is slightly old, but eval_events[id_event][0] seems to be the sampled time for the test , whereas eval_events[id_event][3] uses the true validity range for this event.

We want to filter out in the eval all other events for the pair (lhs, rel) that happened in the validity-range of this eval event no ?

**Comment by ling-cai** [comment]
Yes. i guess this is to filter out some positive predictions to get filtered evaluation metrics. However,
self.events[id_timeline][0] --> this is  the date (start/end) associated with the event id 'id_timeline'.
but I checked on the eval_events[id_event][3],which is the object id. So i am confused. Not sure whether this is a bug or I misunderstood something.

**Comment by timlacroix** [comment]
Actually you're right.
let me fix this and check what happens

**Comment by timlacroix** [comment]
So, this would only affect yago and wikidata results, since this eval is not used for ICEWS.
Results for YAGO do not seem to change, I was able to get 0.35 with the command in the README.md
Evaluating on wikidata would take more time. I will try to see if the results change drastically.
It is very possible that there is very little to filter on these datasets anyway so this fix wouldn't change the results significantly.

**Comment by ling-cai** [comment]
yes. i agree that it would not make a big difference to the results. Just wanted to make sure i understood it correctly. thanks for your quick response.

**Comment by timlacroix** [comment]
So, this would only wikidata results, since this eval is not used for ICEWS.
Evaluating on wikidata would take more time. I will try to see if the results change drastically.
It is very possible that there is very little to filter on these datasets anyway so this fix wouldn't change the results significantly.

**Comment by timlacroix** [comment]
yeah, sorry for the confusion in my answers, it's been a while since I've looked at the code. Thanks for spotting the mistake !
